main_app/igdb_api.py

```diff
@@ -0,0 +1,85 @@
+import math
+
+import requests
+from urllib.parse import urlencode, quote_plus
+
+from django.conf import settings
+
+
+class IGDB:
+    def __init__(self):
+        self.__last_headers_x_count = 1
+        self.__api_key = settings.IGDB_API_KEY
+        self.__api_url = 'https://api-v3.igdb.com/'
+        self.__headers = {
+            'user-key': self.__api_key,
+            'Accept': 'application/json'
+        }
+        self.__limit = 6
+
+    def __api_get(self, additional_string: str = ''):
+        print(f'{self.__api_url}{additional_string}')
+        data = requests.get(f'{self.__api_url}{additional_string}', headers=self.__headers)
+        try:
+            x_count = int(data.headers.get('X-Count'))
+        except TypeError:
+            x_count = 1
+        if x_count > 1:
+            self.__last_headers_x_count = x_count
+        return data.json()
+
+    def api_get_game(self, game_id):
+        data = self.__api_get(f'games/{game_id}?fields=rating,version_title,aggregated_rating,'
+                              f'screenshots,summary,cover,platforms,genres')
+        return data
+
+    def __generate_filters(self, search: str = None, genres: str = None,
+                           platforms: str = None, user_ratings: tuple = None) -> dict:
+        result = {}
+        filters = []
+        if search:
+            result['search'] = search
+        if genres:
+            filters.append(f'[genres][eq]={genres}')
+        if platforms:
+            filters.append(f'[platforms][eq]={platforms}')
+        if user_ratings:
+            try:
+                filters.append(f'[rating][gt]={user_ratings[0] * 10},[rating][lt]={user_ratings[1] * 10}')
+            except IndexError as e:
```

Comment:
it seems that sometimes `user_ratings` list doen't have [0] or [1] elements. Can you use if/else or smth else in order to avoid try/except?